### PR TITLE
HEC-981 adding extra dteTime checked field in tax match Response

### PR DESCRIPTION
**Pull request by priyankadhiman3006**
I have added extra field dateTimeChecked in HECTaxCheckMatchResult as for the valid and Expiry tax check page , we need to display the date time at which the check was done against database. Same will be carried forward in LB.

### CODE REVIEW

File: app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala [hunk 1]
```diff
@@ -70,7 +70,8 @@ class TaxCheckController @Inject() (
             result => Ok(Json.toJson(result))
           )
 
-      case JsError(_) =>
+      case JsError(e) =>
+        logger.warn(s"error is $e")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala:74]: we should take this out I think - if this was in prod we could accidentally log sensitive info

File: app/uk/gov/hmrc/hec/util/TimeUtils.scala [hunk 2]
```diff
@@ -16,12 +16,13 @@
...
-import java.time.{Clock, LocalDate}
+import java.time.{Clock, LocalDate, ZoneId, ZonedDateTime}
 
 object TimeUtils {
 
   val clock: Clock = Clock.systemUTC()
 
-  def today(): LocalDate = LocalDate.now(clock)
+  def today(): LocalDate           = LocalDate.now(clock)
+  def todayByZone(): ZonedDateTime = ZonedDateTime.now(ZoneId.of("Europe/London"))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hec/util/TimeUtils.scala:26]: maybe a name like `now()` or `nowInEuropeLondon()` is more accurate - `today` doesn't include the concept of time. May not need the `InEuropeLondon` part cos the `ZonedDateTime` will allow users to adjust the date time to whatever zone they need to so the base zone is not necessarily required info

File: app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala (hunk 1 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala:74]: Yup, i added for testing my json parsing failure, forgot to remove it . Will do.

File: app/uk/gov/hmrc/hec/util/TimeUtils.scala (hunk 2 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hec/util/TimeUtils.scala:26]: so , def nowByZone(): ZonedDateTime = ZonedDateTime.now()? is this you are trying to say? or ZoneId is still required?

Comment (AndyHWChung) [reply app/uk/gov/hmrc/hec/util/TimeUtils.scala:26]: I meant replacing the name `todayByZone`

File: app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala (hunk 1 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hec/controllers/TaxCheckController.scala]: done

File: app/uk/gov/hmrc/hec/util/TimeUtils.scala (hunk 2 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hec/util/TimeUtils.scala]: done

**Review by AndyHWChung — APPROVED** [review]

**Review by meghana-viswanath — APPROVED** [review]

**Pull request merged**